### PR TITLE
Add a link to "assign-before-let" issue to error message

### PR DESCRIPTION
**Pull request by aprocter-stripe**
Temporary measure to improve the error message in #3482.

### Motivation
We don't have bandwidth right now to implement a complete fix, but it would be nice to direct people to some information on the known bug here.

### Test plan
See included automated tests.

### CODE REVIEW

File: infer/environment.cc [hunk 1]
```diff
@@ -1333,6 +1333,12 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
                                 e.setHeader("Incompatible assignment to variable declared via `{}`: `{}` is not a "
                                             "subtype of `{}`",
                                             "let", tp.type->show(ctx), cur.type->show(ctx));
+                                if (!hasType(ctx, bind.bind.variable) && cur.type.isNilClass()) {
+                                    e.addErrorLine(core::Loc(ctx.file, bind.loc),
+                                                   "This may be caused by a known Sorbet limitation when `{}` is in "
+                                                   "use. See `{}` for more details.",
+                                                   "T.let", "https://github.com/sorbet/sorbet/issues/3482");
```
Comment (elliottt) [inline infer/environment.cc:1340]: This looks great, but do you think we could shorten the message a bit? It looks like it's going to wrap given the test output, perhaps rephrasing it to just "See <issue link> for more details"?

**Review by elliottt — CHANGES_REQUESTED** [review]
This looks great! What do you think about shortening the informational message?

File: infer/environment.cc (hunk 1 above)
Comment (aprocter-stripe) [reply infer/environment.cc:1340]: Sure, should be able to phrase it more economically.

**Comment by aprocter-stripe** [comment]
Hello me.

We have a policy of closing pull requests that have not had any recent activity. If you would still like to see this pull request merged, please feel free to reopen it.

As always, please don't hesitate to reach out here on GitHub or in #internals on [sorbet-slack](https://sorbet.org/slack) if you have any questions.

**Pull request closed without merging**